src/Imgix/UrlHelper.php

```diff
@@ -28,7 +28,14 @@ public function formatPath($path) {
         if (preg_match("/^https?:\/\//", $path)) {
             // If this path is a full URL, encode the entire thing
             $path = rawurlencode($path);
-
+        } else if (preg_match("/^https?:\/\/[^\s\/$.?#]*\.[^\s]*$/", rawurldecode($path))) {
```

Comment:
Do we need the `if` condition above this at all? I might be missing something subtle but I think this new `else if` condition should cover it.

Reply:
The `if` and `else if` handle `%` differently.

If `$path` is `http://mywebsite.com/images/p%C3%BCg.jpg`, `if` block correctly escapes all special characters _including_ `%` (refer `testHelperFormatPathWithFullUrlWithEncodedCharacters` test). However, if the input is already encoded, `else if` encodes special characters _excluding_ `%` (refer test `testHelperFormatPathHttspURLAlreadyEncodedWithSpecialCharacters`).

Reply:
Ahhh, okay got it. I was not recognizing the `[^` in the regex on L36. Thanks for the thorough explanation.